tests/spread/basic/task.yaml

```diff
@@ -0,0 +1,33 @@
+summary: Ensure that things worked
+
+execute: |
+  # for various utilities
+  . "$EXTTESTSLIB/nested.sh"
+
+  # Start the nested UC20 VM
+  start_nested_core_vm_unit
+
+  # At this point we are able to SSH to the nested VM, so things probably worked
+  # but we should wait for snapd to finish seeding anyways, but first we need to
+  # wait for the snap command to exist
+  nested_wait_for_snap_command
+
+  # Now formally wait for seeding
+  execute_remote sudo snap wait system seed.loaded
+
+  # We should be in run mode
+  execute_remote cat /proc/cmdline | MATCH snapd_recovery_mode=run
+
+  # We should have a modeenv
+  execute_remote test -f /var/lib/snapd/modeenv
+
+  # Verify that the partitions we expect, are present
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-seed | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-boot | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-save | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-data | xargs test
+  
+  # Verify system services are running
+  execute_remote systemctl -q is-active snapd
+  execute_remote snap change 1
```

Comment:
I don't see any cleanup for the test, to make sure the test is cleaning up correctly the environment you can run spread with repeat like this:

spread -debug -repeat 10 google-nested:ubuntu-20.04-64:tests/spread/basic

Reply:
There is cleanup in the spread.yaml with the restore-each section. Is this sufficient or should there be more?

Reply:
I just ran your command and it seems to be just fine. It runs the restore action from spread.yaml